core/store/models/service_agreement_test.go

```diff
@@ -87,36 +98,87 @@ func TestEncumbrance_ABI(t *testing.T) {
+			"0x" +
+				"0000000000000000000000000000000000000000000000000000000000000000" +
+				"0000000000000000000000000000000000000000000000000000000000000000" + "886e0900" +
+				"0000000000000000000000000000000000000000000000000000000000000000" + // Aggregator address
+				"00000000" + "00000000", // Function selectors
+		},
 		{"oracle address", nil, 0, models.AnyTime{},
-			[]models.EIP55Address{models.EIP55Address("0xa0788FC17B1dEe36f057c42B6F373A34B014687e")},
-			"0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000886e0900000000000000000000000000a0788fc17b1dee36f057c42b6f373a34b014687e"},
+			[]models.EIP55Address{models.EIP55Address("0xa0788FC17B1dEe36f057c42B6F373A34B014687e")}, "", "", "",
+			"0x" +
+				"0000000000000000000000000000000000000000000000000000000000000000" +
+				"0000000000000000000000000000000000000000000000000000000000000000" + "886e0900" +
+				"000000000000000000000000a0788fc17b1dee36f057c42b6f373a34b014687e" + // Oracle address
+				"0000000000000000000000000000000000000000000000000000000000000000" + // Aggregator address
+				"00000000" + "00000000", // Function selectors
+		},
 		{"different endAt", nil, 0, models.NewAnyTime(endAt),
-			[]models.EIP55Address{models.EIP55Address("0xa0788FC17B1dEe36f057c42B6F373A34B014687e")},
-			"0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000459a7465000000000000000000000000a0788fc17b1dee36f057c42b6f373a34b014687e"},
+			[]models.EIP55Address{models.EIP55Address("0xa0788FC17B1dEe36f057c42B6F373A34B014687e")}, "", "", "",
+			"0x" +
+				"0000000000000000000000000000000000000000000000000000000000000000" +
+				"0000000000000000000000000000000000000000000000000000000000000000" + "459a7465" +
+				"000000000000000000000000a0788fc17b1dee36f057c42b6f373a34b014687e" +
+				"0000000000000000000000000000000000000000000000000000000000000000" + // Aggregator address
+				"00000000" + "00000000", // Function selectors
+		},
+		{name: "aggregator info", payment: nil, expiration: 0, endAt: models.NewAnyTime(endAt),
+			oracles: []models.EIP55Address{
+				models.EIP55Address("0xa0788FC17B1dEe36f057c42B6F373A34B014687e"),
+			},
+			aggregator:             "0x3141592653589793238462643383279502884197",
+			aggInitiateJobSelector: "0x12345678", aggFulfillSelector: "0x87654321",
+			want: "0x" +
+				"0000000000000000000000000000000000000000000000000000000000000000" +
+				"0000000000000000000000000000000000000000000000000000000000000000" + "459a7465" +
+				"000000000000000000000000a0788fc17b1dee36f057c42b6f373a34b014687e" + // Oracle address
+				"0000000000000000000000003141592653589793238462643383279502884197" + // Aggregator address
+				"12345678" + "87654321", // Function selectors
+		},
 	}
 
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
+			fs := func(s string) models.FunctionSelector {
+				if s == "" {
```

Comment:
This statement never appears to be true.

Reply:
Thanks. Removed in 3c3f149a16e9fa53fd5c8567ba32f8f9fd875d34

Reply:
Oops, actually this is used, because the selectors specified in the tests are empty. I've restored this in c061ce4d7bfa639b072c9e7c00acee09ef8da8b7. (I'll squash before I merge.)

Reply:
Rather than using an if statement here, I think it would be clearer to put the empty selector in the test definition.